### ISSUE TITLE
Query arguments for level project when retrieving robot users using the API does nothing

### ISSUE THREAD

**Issue by flaviomoringa** [goharbor/harbor#16229]
**Expected behavior and actual behavior:**
I want to automate, by using the API, the creation of some needed robot users per project. For that I first want to check if the project already has the needed robot account.

The API that returns the list of Robot accounts only outputs the Robot system accounts, even if I change the query parameters.

I want to be able to list all the robot accounts for a given project.

There seems to be another issue that is when using the query "name". It seems it just expects the name tag with no prefix, which makes it very strange to use, because the API returns the name with the prefix, which makes all of this very inconsistent.

**Steps to reproduce the problem:**

I'm retrieving the robot users using the API like this:
curl -s -u admin:XXXXXXXXX -X GET "https://harbor.mydomain.com/api/v2.0/robots?&page=1&page_size=0" -H "accept: application/json"

This works, and lists all my System robot accounts.

Now I'm trying with the query parameter to list  all robot users from the projects level:

curl -s -u admin:XXXXXXXXX -X GET "https://harbor.mydomain.com/api/v2.0/robots?&page=1&page_size=0&q=level=project" -H "accept: application/json"

And it just shows again the system robot accounts. I have at least one project with a robot account.

The API help does say anything about this.

**Versions:**
Please specify the versions of following systems.

- harbor version: 2.4.0

**Comment by zyyw** [comment]
please try a url like this one below:
```
YOUR_HARBOR_DOMAIN/api/v2.0/robots?q=Level=project,ProjectID=1&page_size=15&page=1
```
and replace the projectID with the one of your project ID.

**Comment by flaviomoringa** [comment]
@zyyw thank you very much, that worked.

I also managed to do a search like this:
YOUR_HARBOR_DOMAIN/api/v2.0/robots?q=Level=project,ProjectID=1,name=~jenkins,&page_size=15&page=1

With you help I now understood that the swagger API documentation is not very accurate, and the API itself not that consistent.

The query params need to be:
"Level", not "level"
"ProjectID" not "projectid"
But "name" is "name"

But now I know.

However this should be fixed, since I spent quite a lot of time trying to have this working, when it would just be a matter of proper documentation  :-( 

I will leave it open, since the documentation does need fixing, but if any of the developers prefer that I open a new ticket just let me know.

Thanks again.
